tools/codegen/model_generator/model.go

```diff
@@ -0,0 +1,257 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model_generator
```

Comment:
please rename this package. there should not be underscores in the package name.

coming up with a new name is tough, given the use of `Model` below. I'd suggest `model`, but `model.Model` is a bit stutter-y.  maybe we call this package `model` and the struct `Generated`  (for `model.Generated`)  ?